util/dvsim/SimCfg.py

```diff
@@ -236,6 +196,46 @@ def __post_init__(self):
         # Run some post init checks
         super().__post_init__()
 
+
+    def _resolve_wave_format(self):
+        '''Decide on the correct wave format if waves are enabled.
+
+        This is called after reading the config file. If waves is None,
+        then simply exit, since there is nothing to do here. The tool which
+        must be set at this point, supports a limited list of formats. If waves
+        is set to 'default', then pick the first item on the list. If not,
+        waves is set to the desired format. In that case we need to check if
+        the tool actually supports it.
+        '''
```

Comment:
I don't think this comment is quite correct for the None case: you convert `None` to `'none'` (presumably so that it serialises nicely to hjson). That probably needs documenting. It's kind of a pity that we re-use the `self.waves` variable and change its type like this. I wonder whether there's a cleaner way to do it.

Obviously, this doesn't block the PR, but I'm not sure that your change from a static function to a method improves things either. Reading the call site no longer shows instantly what the function does (set `self.waves`). In general, I'd recommend returning values rather than mutating in place: it makes the code much easier to read.

Reply:
I definitely agree about the argument changes: they're a great improvement, thanks! And thanks for changing the shape of the function. I think it's probably a bit easier to understand this way around.